nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/PersistenceContext.java

```diff
@@ -1596,8 +1599,9 @@ public void restoreVersion(Node node, Node version) {
         Serializable versionableId = node.getId();
         Serializable versionId = version.getId();
 
-        // clear complex properties
-        List<SimpleFragment> children = getChildren(versionableId, null, true);
+        // clear complex properties and exclude special children to avoid duplicates on restore
+        boolean excludeSpecialChildren = true;
+        List<SimpleFragment> children = getChildren(versionableId, null, excludeSpecialChildren);
```

Comment:
The third arg of `getChildren` is `complexProp` so it's very misleading to pass it an arg that's called `excludeSpecialChildren`. I'm not sure why this needs to get changed at all to be honest.

Reply:
I was renaming my arguments and mistook this one with one with one of the flags I introduced